### PR TITLE
Feature/layered architectures module 2 task 2

### PR DESCRIPTION
**Pull request by oleksii-yurchenko-easypark**

### CODE REVIEW

File: layered-architectures-module-2/Catalog.Application/Categories/Commands/UpdateCategory/UpdateCategoryCommand.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    public int Id { get; set; }
+    
+    public string Name { get; set; }
+    
+    public string ImageUrl { get; set; }
+    
+    public int? ParentCategoryId { get; set; }
+}
+
+public class UpdateCategoryCommandHandler : IRequestHandler<UpdateCategoryCommand>
```
Comment (anton-havrenkov) [inline layered-architectures-module-2/Catalog.Application/Categories/Commands/UpdateCategory/UpdateCategoryCommand.cs]: this Handler could be moved inside UpdateCategoryCommand

File: layered-architectures-module-2/Catalog.Application/Categories/Commands/UpdateCategory/UpdateCategoryCommand.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+using Catalog.Application.Common.Interfaces;
+using Catalog.Application.Common.Mappings;
+using Catalog.Domain.Entities;
+using MediatR;
+
+namespace Catalog.Application.Categories.Commands.UpdateCategory;
+
+public class UpdateCategoryCommand : IRequest, IMapFrom<Category>
+{
+    public int Id { get; set; }
```
Comment (anton-havrenkov) [inline layered-architectures-module-2/Catalog.Application/Categories/Commands/UpdateCategory/UpdateCategoryCommand.cs]: it's better to have a separate DTO model for data. So you could extend data and logic (command might contains configuration data) separately

Comment (anton-havrenkov) [reply layered-architectures-module-2/Catalog.Application/Categories/Commands/UpdateCategory/UpdateCategoryCommand.cs]: You already have CategoryDto for that

File: layered-architectures-module-2/Catalog.Application/Products/Commands/CreateProduct/CreateProductCommandValidator.cs [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+        RuleFor(p => p.Name)
+            .NotEmpty().WithMessage("Name is required.")
+            .MaximumLength(50).WithMessage("Name must not exceed 50 characters.");
+
+        RuleFor(p => p.CategoryId).NotEqual(0);
+
+        RuleFor(p => p.Price).NotEqual(0);
+        
+        RuleFor(p => p.Amount).NotEqual((uint)0);
```
Comment (anton-havrenkov) [inline layered-architectures-module-2/Catalog.Application/Products/Commands/CreateProduct/CreateProductCommandValidator.cs]: must be greater > 0

File: layered-architectures-module-2/Catalog.Application/Products/Commands/CreateProduct/CreateProductCommandValidator.cs [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    {
+        _context = context;
+
+        RuleFor(p => p.Name)
+            .NotEmpty().WithMessage("Name is required.")
+            .MaximumLength(50).WithMessage("Name must not exceed 50 characters.");
+
+        RuleFor(p => p.CategoryId).NotEqual(0);
+
+        RuleFor(p => p.Price).NotEqual(0);
```
Comment (anton-havrenkov) [inline layered-architectures-module-2/Catalog.Application/Products/Commands/CreateProduct/CreateProductCommandValidator.cs]: must be greater > 0

File: layered-architectures-module-2/Catalog.Application/Products/Commands/CreateProduct/CreateProductCommandValidator.cs [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+    public CreateProductCommandValidator(IApplicationDbContext context)
+    {
+        _context = context;
+
+        RuleFor(p => p.Name)
+            .NotEmpty().WithMessage("Name is required.")
+            .MaximumLength(50).WithMessage("Name must not exceed 50 characters.");
+
+        RuleFor(p => p.CategoryId).NotEqual(0);
```
Comment (anton-havrenkov) [inline layered-architectures-module-2/Catalog.Application/Products/Commands/CreateProduct/CreateProductCommandValidator.cs]: must match to some existing category

File: layered-architectures-module-2/Catalog.Application/Products/Commands/DeleteProduct/DeleteProductCommand.cs [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    public DeleteProductCommandHandler(IApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<Unit> Handle(DeleteProductCommand request, CancellationToken cancellationToken)
+    {
+        var entity = await _context.Products
+            .Where(l => l.Id == request.Id)
+            .SingleOrDefaultAsync(cancellationToken);
```
Comment (anton-havrenkov) [inline layered-architectures-module-2/Catalog.Application/Products/Commands/DeleteProduct/DeleteProductCommand.cs]: do you know the diff between SingleOrDefault vs FirstOrDefaultAsync? 
do you know FindAsync?

Comment (oleksii-yurchenko-easypark) [reply layered-architectures-module-2/Catalog.Application/Products/Commands/DeleteProduct/DeleteProductCommand.cs]: You're right, I considered to use FindAsync, suit better here

**Review by anton-havrenkov — APPROVED** [review]

**Review by AlexeyUmka — APPROVED** [review]

**Pull request merged**